Annotato/Annotato/Views/Document/List/DocumentListViewController.swift

```diff
@@ -43,8 +47,28 @@ class DocumentListViewController: UIViewController {
     }
 }
 
-extension DocumentListViewController: DocumentListToolbarDelegate, Navigable {
-    func didTapAddButton() {
-        goToDocumentEdit()
+extension DocumentListViewController: DocumentListToolbarDelegate,
+        UIDocumentPickerDelegate,
+        DocumentListCollectionCellViewDelegate,
+        Navigable {
+
+    func didTapImportFileButton() {
+        goToImportingFiles()
+    }
+
+    func documentPicker(_ controller: UIDocumentPickerViewController, didPickDocumentsAt baseFileUrls: [URL]) {
+        guard let selectedFileUrl = baseFileUrls.first else {
+            return
+        }
+        guard FileManager.default.urls(for: .documentDirectory, in: .userDomainMask).first != nil else {
+            return
+        }
+        let newlyLoadedDocumentPdfViewModel = DocumentPdfViewModel(baseFileUrl: selectedFileUrl)
+        goToDocumentEdit(documentPdfViewModel: newlyLoadedDocumentPdfViewModel)
+    }
+
+    func didSelectCellView(document: DocumentListViewModel) {
+        let selectedDocumentPdfViewModel = DocumentPdfViewModel(baseFileUrl: document.baseFileUrl)
+        goToDocumentEdit(documentPdfViewModel: selectedDocumentPdfViewModel)
```

Comment:
I just realised that it shouldn't be the responsibility of goToDocumentEdit to decide how to create a new view model.

can you replace it with

```suggestion
        let selectedDocumentPdfViewModel = DocumentPdfViewModel(baseFileUrl: document.baseFileUrl)
        let documentViewModel = DocumentViewModel(
            annotations: [], // Since there are no annotations at point of upload
            pdfDocument: documentPdfViewModel
        )
        goToDocumentEdit(documentViewModel: documentViewModel)
```

Reply:
I think it has to be there because I need access to the view controller to set the currentDocumentViewModel
Otherwise the goToDocumentEdit needs to return the view controller which is not ideal

Reply:
@darren2pro As in instead of passing in `documentPdfViewModel`, you pass in `documentViewModel`. You will still be setting currentDocumentViewModel inside goToDocumentEdit. I'll change it for you and merge this.

Reply:
Besides it's wrong to create `documentViewModel` based on the pdf inside the navigation. How are you supposed to get the other attributes of `documentViewModel` in this way??